cms/djangoapps/contentstore/views/utilities/bulkupdate.py

```diff
@@ -0,0 +1,245 @@
+        kwargs_for_reverse.update(kwargs)
+    return reverse('contentstore.views.' + handler, kwargs=kwargs_for_reverse)
+
+
+def _save_request_status(request, key, status):
+    """
+    Internal method used to save update status for a course in request
+    session
+    """
+    session_status = request.session.get('update_status')
+    if session_status is None:
+        session_status = request.session.setdefault("update_status", {})
+
+    session_status[key] = status
+    request.session.save()
+
+
+def _update_advanced_settings(request, course_key_string, session_status_string, modified_settings):
+    """
+    Internal method used to save to advanced settings
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    course = modulestore().get_course(course_key, 3)
+    try:
+        with modulestore().bulk_operations(course_key):
+            for key, value in modified_settings.iteritems():
+                setattr(course, key, value)
+
+            modulestore().update_item(course, request.user.id)
+
+    except:
+        _save_request_status(request, session_status_string, 2)
+        return JsonResponse(
+            {
+                'ErrMsg': 'Unable to update advanced settings',
+                'Stage': -2
+            },
+            status=500
+        )
+
+
+def _utility_bulkupdate_get_handler(request, course_key_string):
+    """
+    Internal bulkupdate handler for GET operation
+    """
+    course_outline_url = _reverse_course_url(
+        "course_handler",
+        course_key_string
+    )
+    update_url = _reverse_course_url(
+        "utility_bulkupdate_handler",
+        course_key_string
+    )
+    status_url = _reverse_course_url(
+        "utility_bulkupdate_status_handler",
+        course_key_string,
+        kwargs={
+            'max_attempts': 'fillerMaxAttempts',
+            'show_answer': 'fillerShowAnswer'
+        }
+    )
+
+    course_key = CourseKey.from_string(course_key_string)
+    course = modulestore().get_course(course_key, 3)
+
+    return render_to_response(
+        'bulkupdate.html',
+        {
+            'context_course': course,
+            'course_outline_url': course_outline_url,
+            'bulkupdate_update_url': update_url,
+            'bulkupdate_status_url': status_url,
+            'default_max_attempts': DEFAULT_MAX_ATTEMPTS,
+            'default_show_answer': DEFAULT_SHOW_ANSWER,
+            'show_answer_options': SHOW_ANSWER_OPTIONS
+        }
+    )
+
+
+def _utility_bulkupdate_post_handler(request, course_key_string):
+    """
+    Internal bulkupdate handler for POST operation
+    """
+    modified_settings = {}
+    user = request.user
+
+    # Retrieve settings from request
+    try:
+        max_attempts = request.POST.get('maxAttempts')
+        show_answer = request.POST.get('showAnswer')
+        session_status = request.session.setdefault("update_status", {})
+        session_status_string = course_key_string + max_attempts + show_answer
```

Comment:
You probably need a delimiter in there.

Consider:
```python
'Stanford/Course/Fall2017` + 10 + True
```

vs

```python
'Stanford/Course/Fall20171` + 0 + True
```

Reply:
How about this? 

`return 'course_key_string={}&max_attempts={}&show_answer={}'.format(
        course_key_string,
        max_attempts,
        show_answer,
    )`